identity_stardust/src/network/network_name.rs

```diff
@@ -0,0 +1,147 @@
+// Copyright 2020-2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use std::borrow::Cow;
+
+use core::convert::TryFrom;
+use core::fmt::Display;
+use core::fmt::Formatter;
+use core::ops::Deref;
+use std::fmt::Debug;
+
+use serde::Deserialize;
+use serde::Serialize;
+
+use crate::error::Error;
+use crate::error::Result;
+
+/// Network name compliant with the [`crate::StardustDID`] method specification.
+#[derive(Clone, Hash, PartialEq, Eq, PartialOrd, Ord, Deserialize, Serialize)]
+#[repr(transparent)]
+pub struct NetworkName(Cow<'static, str>);
+
+impl NetworkName {
+  pub(crate) const MAX_LENGTH: usize = 6;
+  pub(crate) const DEFAULT_STR: &'static str = "main";
+  // Names of networks maintained by the IF that the stardust UTXO based DID method should support
+  const IOTA_NETWORK_NAMES: [&'static str; 4] = [Self::DEFAULT_STR, "dev", "smr", "rms"];
```

Comment:
This isn't really a concern of `NetworkName`, but whatever uses it. These names have to be mapped to endpoints to be useful.

```suggestion
```

Reply:
Was mainly meant as a way to optimize the `validate_network_name` function, but I never measured the impact of doing this so it can be removed for now.